LibCarla/source/carla/trafficmanager/CollisionStage.cpp

```diff
@@ -261,8 +261,8 @@ namespace CollisionStageConstants {
       // Whichever vehicle's path is farthest away from the other vehicle gets priority to move.
       if (geodesic_path_bbox_touching
           && ((!vehicle_bbox_touching
-           && (!ego_path_clear || (ego_path_clear && other_path_clear && !ego_path_priority)))
-              || (vehicle_bbox_touching && !ego_angular_priority))) {
+           && (!ego_path_clear || (ego_path_clear && other_path_clear && !ego_angular_priority && !ego_path_priority)))
```

Comment:
More stringent conditions for stopping decisions. Works well to alleviate deadlocks in roundabout turns in testing.